azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -606,6 +912,112 @@ func flattenEventGridEventSubscriptionSubjectFilter(filter *eventgrid.EventSubsc
 	return []interface{}{result}
 }
 
+func flattenEventGridEventSubscriptionAdvancedFilter(input *[]eventgrid.BasicAdvancedFilter) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	boolEquals := make([]interface{}, 0)
+	numberGreaterThan := make([]interface{}, 0)
+	numberGreaterThanOrEquals := make([]interface{}, 0)
+	numberLessThan := make([]interface{}, 0)
+	numberLessThanOrEquals := make([]interface{}, 0)
+	numberIn := make([]interface{}, 0)
+	numberNotIn := make([]interface{}, 0)
+	stringBeginsWith := make([]interface{}, 0)
+	stringEndsWith := make([]interface{}, 0)
+	stringContains := make([]interface{}, 0)
+	stringIn := make([]interface{}, 0)
+	stringNotIn := make([]interface{}, 0)
```

Comment:
since the `key`, `value` and `values` properties can be nil/omitted from the Azure API here - it'd be worth adding a couple of inline functions to assist:

```
var flattenValue = func(inputKey, inputValue *string) map[string]interface{} {
  key := ""
  if inputKey != null {
    key = *inputKey
  }
  value := ""
  if inputValue != null {
    value = *inputValue
  }

  return map[string]interface{}{
    "key": key,
    "value": value,
  }
}
var flattenValues = func(inputKey *string, inputValues *[]string) map[string]interface{} {
  key := ""
  if inputKey != null {
    key = *inputKey
  }
  values := make([]string, 0)
  if inputValues != null {
    values = *inputValues
  }

  return map[string]interface{}{
    "key": key,
    "values": values,
  }
}
```